data/transactions/logic/assembler.go

```diff
@@ -786,60 +778,73 @@ func assembleSubstring(ops *OpStream, spec *OpSpec, args []string) error {
 	return nil
 }
 
-func assembleTxn(ops *OpStream, spec *OpSpec, args []string) error {
-	if len(args) != 1 {
-		return ops.error("txn expects one argument")
-	}
-	fs, ok := txnFieldSpecByName[args[0]]
+func txnFieldImm(name string, expectArray bool, ops *OpStream) (*txnFieldSpec, error) {
+	fs, ok := TxnFieldSpecByName[name]
 	if !ok {
-		return ops.errorf("txn unknown field: %#v", args[0])
+		return nil, fmt.Errorf("unknown field: %#v", name)
```

Comment:
`name` is a string. is there any particular reason to avoid using `%s` here ?

Reply:
That's an easy way to get it quoted, which I think makes a nicer error message.

Reply:
The reason I'm trying to avoid using `#v` when not needing it is that is would call the `String()` method, which, in turn, might have some unexpected outcome. We had a bug where a similar case was leading to infinite recursive call ( i.e. the String() method was using format, which ended up with recursive call )

Reply:
%s also called the String() method if it exists.